tokio/src/sync/mutex.rs

```diff
@@ -10,9 +10,11 @@ use std::sync::Arc;
 
 /// An asynchronous `Mutex`-like type.
 ///
-/// This type acts similarly to an asynchronous [`std::sync::Mutex`], with one
-/// major difference: [`lock`] does not block and the lock guard can be held
-/// across await points.
+/// This type acts similarly to [`std::sync::Mutex`], with two major
+/// differences: [`lock`] is an async method so does not block, and the lock
+/// guard is [`Send`] so can be held across any kind of `.await` point, not just
+/// those which keep the task within a single thread (such as in the context of
+/// a local task set or within a `join!` on the main thread, etc.)
```

Comment:
We should never encourage the user to hold a blocking mutex across an await point, even if the compiler permits it due to the task not being movable across threads. This is because it can incredibly easily lead to deadlocks.

Reply:
That is, although the compiler sometimes permits it, it is never correct to do so.

Reply:
Yeah I see your point. I have pushed another edit. It's slightly more verbose but I think there are benefits to the new vesion for less experienced readers.